consensus/reactor.go

```diff
@@ -272,21 +272,57 @@ func (conR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
 			conR.conS.mtx.Lock()
 			initialHeight := conR.conS.state.InitialHeight
 			conR.conS.mtx.Unlock()
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(msg.Step),
+				string(e.Src.ID()),
+				schema.ConsensusNewRoundStep,
+				schema.Download,
+			)
 			if err = msg.ValidateHeight(initialHeight); err != nil {
 				conR.Logger.Error("Peer sent us invalid msg", "peer", e.Src, "msg", msg, "err", err)
 				conR.Switch.StopPeerForError(e.Src, err)
 				return
 			}
 			ps.ApplyNewRoundStepMessage(msg)
 		case *NewValidBlockMessage:
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(ps.PRS.Step),
```

Comment:
I wasn't sure what do include this as. If we want to be consistent, then we encounter a data race due to how the consensus reactor manages state. Perhap the better solution here is to have different tables instead of sticking everything in one

Reply:
The consistency of the data is crucial since it will serve as the foundation for our analysis. If this issue warrants a separate PR, I suggest we temporarily remove the step parameter and consider adding it in a follow-up PR.

Reply:
the above solution isn't good and is confusing, however I would argue that its still consistent, since each individual type of consensus state is handled the same through out. It's just that we are using the same struct for different types of consensus state.

that being said, this was handled in a more explicit way in https://github.com/celestiaorg/celestia-core/pull/1292/commits/88700d5e01a930e33e9e12fa6d5ed3c239f67589